flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer.java

```diff
@@ -683,13 +682,13 @@ public void close() throws FlinkKafkaException {
 		} finally {
 			// We may have to close producer of the current transaction in case some exception was thrown before
 			// the normal close routine finishes.
-			if (currentTransaction() != null) {
-				IOUtils.closeQuietly(currentTransaction().producer);
-			}
-			// Make sure all the producers for pending transactions are closed.
-			pendingTransactions().forEach(transaction ->
-					IOUtils.closeQuietly(transaction.getValue().producer)
-			);
+//			if (currentTransaction() != null) {
```

Comment:
If we comment this code snippet then the test case will be passed. Otherwise, the log will display this log message `Aborting incomplete transaction due to shutdown ...` and the transcation state would transform from ongoing to `PrepareAbort`.

One idea is to get the ` TransactionManager.State` from `TransactionManager` in Kafka client before closing the producer. WDYT? @aljoscha and @pnowojski